### PR TITLE
Only delete dag runs/task instances during testing that match pattern

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes a testing issue identified by @stacimc

## Description

After the merging of #636, it appeared that our tests were failing on `main`. Further investigation showed that tests would run fine synchronously (e.g. `just test -n 1`) but would fail then run in parallel. I was able to track this down to the `test_single_run_external_dags_sensor.py` tests, which were deleting all `DagRun`s and `TaskInstances` as part of setup/teardown. This PR changes those deletions to match a specific DAG prefix, which is used when creating DAGs throughout the test.

## Testing Instructions

`just test` should be sufficient.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: tests/dags/common/sensors/test_single_run_external_dags_sensor.py [hunk 1]
```diff
@@ -17,13 +17,20 @@
...
+DAG_PREFIX = "sreds"  # single_run_external_dags_sensor
 
 
 @pytest.fixture(autouse=True)
 def clean_db():
     with create_session() as session:
-        session.query(DagRun).delete()
-        session.query(TaskInstance).delete()
+        # synchronize_session='fetch' required here to refresh models
+        # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
```
Comment (krysal) [inline tests/dags/common/sensors/test_single_run_external_dags_sensor.py:27]: Noticing now that SO uses a CC license! I'll keep that in mind for when I copy links too 💯

**Review by krysal — APPROVED** [review]
I couldn't get an error running the tests on main with `-n 2|3` but the test looks good to me!

File: tests/dags/common/sensors/test_single_run_external_dags_sensor.py (hunk 1 above)
Comment (AetherUnbound) [reply tests/dags/common/sensors/test_single_run_external_dags_sensor.py:27]: I saw someone else do it (I think maybe @sarayourfriend?) and I loved the idea 😄

**Review by stacimc — APPROVED** [review]
I am able to reproduce this one pretty consistently for whatever reason, and can confirm that the fix works great for me! Nice one 💯

**Pull request merged**